src/js/utils/__tests__/deviceFeaturesUtils.test.js

```diff
@@ -176,4 +180,37 @@ describe('utils/deviceFeaturesUtils', () => {
         // without capabilities
         expect(getUnavailableCapabilities({}, coins, support)).toEqual({});
     });
+
+    it('parseRevision', () => {
+        const featT1 = {
+            revision: 'f4424ece1ccb7fc0d6cad00ff840fac287a34f07',
+        };
+        const featT2 = {
+            revision: '6466303936336563',
+        };
+        const parsedRevisionT1 = 'f4424ece1ccb7fc0d6cad00ff840fac287a34f07';
+        const parsedRevisionT2 = 'df0963ec';
+
+        const featBootloader = {
+            revision: null,
+        };
+        const parsedRevisionBootloader = null;
+
+        const featRevisionOnlyNumbers = {
+            revision: '2442434213337100161230033840333287234307',
+        };
+        const parsedRevisionOnlyNumbers = '2442434213337100161230033840333287234307';
+
+        // T1
+        expect(parseRevision(featT1)).toEqual(parsedRevisionT1);
+
+        // T2
+        expect(parseRevision(featT2)).toEqual(parsedRevisionT2);
+
+        // bootloader
+        expect(parseRevision(featBootloader)).toEqual(parsedRevisionBootloader);
+
+        // UTF-8 T1 revision with only numbers
+        expect(parseRevision(featRevisionOnlyNumbers)).toEqual(parsedRevisionOnlyNumbers);
+    });
```

Comment:
It seems to me that we're testing 3 scenarios here each with distinct and independent setup and evaluation phases. The test name is also quite ambiguous. Could we split this to 3 separate tests each with a descriptive name? Something like
```suggestion
    it('parses T1 revision', () => {
        const features = {
            revision: 'f4424ece1ccb7fc0d6cad00ff840fac287a34f07',
        };
        const result = 'f4424ece1ccb7fc0d6cad00ff840fac287a34f07';
        
        expect(parseRevision(features)).toEqual(result);
    })
    
    it('parses T2 revision', () => {
        ...
    });
    
    it('null is passed through', () => {
        ...
    });
    
    it('parses T1 revision containing only numbers', () => {
        ...
    });
```

When such test fail it will be much clearer what went wrong on the first sight. .)

Reply:
It's true that I did sloppy work here. I was inspired by the other tests in this file. I'll fix it.

Reply:
Btw the tests can be basically one-liners if we omit the setup expressions and inline them into the `expect` invocations. .)

```js
it('parses T1 revision', () => {
    expect(parseRevision({revision: 'f4424ece1ccb7fc0d6cad00ff840fac287a34f07'))
        .toEqual('f4424ece1ccb7fc0d6cad00ff840fac287a34f07');
})

it('null is passed through', () => {
    expect(parseRevision({revision: null)).toBeNull();
})
```